discovery/zookeeper/zookeeper.go

```diff
@@ -21,6 +21,24 @@ func init() {
 	discovery.Register("zk", &ZkDiscoveryService{})
 }
 
+func (s *ZkDiscoveryService) fullpath() string {
+	return "/" + strings.Join(s.path, "/")
+}
+
+func (s *ZkDiscoveryService) createFullpath() error {
+	for i := 1; i <= len(s.path); i++ {
+		newpath := "/" + strings.Join(s.path[:i], "/")
+		_, err := s.conn.Create(newpath, []byte{1}, 0, zk.WorldACL(zk.PermAll))
```

Comment:
@aluzzardi can you share the thoughts of `[]byte{1}` here? 
I doubt of the fix in https://github.com/docker/swarm/issues/974